planning/behavior_path_planner/include/behavior_path_planner/planner_manager.hpp

```diff
@@ -0,0 +1,257 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_PATH_PLANNER__PLANNER_MANAGER_HPP_
+#define BEHAVIOR_PATH_PLANNER__PLANNER_MANAGER_HPP_
+
+#include "behavior_path_planner/scene_module_v2/scene_module_interface.hpp"
+#include "behavior_path_planner/scene_module_v2/scene_module_manager_interface.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <autoware_auto_planning_msgs/msg/path_with_lane_id.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <unique_identifier_msgs/msg/uuid.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace behavior_path_planner
+{
+
+using autoware_auto_planning_msgs::msg::PathWithLaneId;
+using tier4_autoware_utils::generateUUID;
+using tier4_autoware_utils::StopWatch;
+using unique_identifier_msgs::msg::UUID;
+using ModuleID = std::pair<std::shared_ptr<SceneModuleManagerInterface>, UUID>;
+
+struct SceneModuleStatus
+{
+  explicit SceneModuleStatus(const std::string & n) : module_name(n) {}
+
+  std::string module_name;
+
+  bool is_execution_ready{false};
+  bool is_waiting_approval{false};
+
+  ModuleStatus status{ModuleStatus::SUCCESS};
+};
+
+class PlannerManager
+{
+public:
+  PlannerManager(rclcpp::Node & node, const bool verbose);
+
+  BehaviorModuleOutput run(const std::shared_ptr<PlannerData> & data);
+
+  void registerSceneModuleManager(
+    const std::shared_ptr<SceneModuleManagerInterface> & scene_module_manager_ptr)
+  {
+    RCLCPP_INFO(logger_, "register %s module", scene_module_manager_ptr->getModuleName().c_str());
+    scene_manager_ptrs_.push_back(scene_module_manager_ptr);
+    processing_time_.emplace(scene_module_manager_ptr->getModuleName(), 0.0);
+  }
+
+  void updateModuleParams(const std::vector<rclcpp::Parameter> & parameters)
+  {
+    std::for_each(
+      scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(),
+      [&parameters](const auto & m) { m->updateModuleParams(parameters); });
+  }
+
+  void reset()
+  {
+    approved_modules_.clear();
+    candidate_module_id_ = boost::none;
+    root_lanelet_ = boost::none;
+    std::for_each(
+      scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(), [](const auto & m) { m->reset(); });
+    resetProcessingTime();
+  }
+
+  void publishDebugMarker()
+  {
+    std::for_each(scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(), [](const auto & m) {
+      m->publishDebugMarker();
+    });
+  }
+
+  std::vector<std::shared_ptr<SceneModuleManagerInterface>> getSceneModuleManagers() const
+  {
+    return scene_manager_ptrs_;
+  }
+
+  std::vector<std::shared_ptr<SceneModuleStatus>> getSceneModuleStatus() const
+  {
+    std::vector<std::shared_ptr<SceneModuleStatus>> ret;
```

Comment:
It seems we can reserve this vector in advance.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2919/commits/e907f5c0e12d36a96cfaa72df24327fd41c74e48.